frontend/src/components/module-editor/BlockPreview.tsx

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { ContentType } from "../../types/ModuleEditorTypes";
+
+const BlockPreview = ({ type }: { type: ContentType }): React.ReactElement => {
+  let title = "";
+
+  switch (type) {
+    case ContentType.text:
```

Comment:
Thoughts on giving the ContentType enum a text string name? Whether that's an attribute or the auto-generated name

Reply:
Do you mean storing the shown name as the value of the enum? If so, I'm unsure how that would scale if we also wanted to add preview images to each thing. I was thinking the switch case was the most versitile.

Reply:
Switching to a class based implementation.